espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/Screenshot.java

```diff
@@ -0,0 +1,88 @@
+package io.appium.espressoserver.lib.handlers;
+
+import android.graphics.Bitmap;
+import android.os.Environment;
+import android.util.Base64;
+import android.view.View;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Date;
+
+
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException;
+import io.appium.espressoserver.lib.model.AppiumParams;
+import io.appium.espressoserver.lib.viewaction.RootViewFinder;
+
+public class Screenshot implements RequestHandler<AppiumParams, String> {
+
+    @Override
+    public String handle(AppiumParams params) throws AppiumException {
+        return takeScreenshot();
+    }
+
+
+    private String takeScreenshot() throws AppiumException {
+        Date now = new Date();
+        android.text.format.DateFormat.format("yyyy-MM-dd_hh:mm:ss", now);
+
+        try {
+            // image naming and path  to include sd card  appending name you choose for file
+            String mPath = Environment.getExternalStorageDirectory().toString() + "/" + now + ".jpg";
+
+            // create bitmap screen capture
+            View v1 = (new RootViewFinder()).getRootView();
+            v1.setDrawingCacheEnabled(true);
+            Bitmap bitmap = Bitmap.createBitmap(v1.getDrawingCache());
+            v1.setDrawingCacheEnabled(false);
+
+            File imageFile = new File(mPath);
+
+
+            FileOutputStream outputStream = new FileOutputStream(imageFile);
+            int quality = 100;
+            bitmap.compress(Bitmap.CompressFormat.JPEG, quality, outputStream);
+
+            outputStream.flush();
+            outputStream.close();
+            return encodeFileToBase64Binary(imageFile);
+        } catch (Throwable e) {
```

Comment:
Newer catch **Throwable** explicitly, since it includes OutOfMemory exception and other ones, that we don't want to intercept for sure. In general case it is enough to catch just **Exception**.